src/win/pipe.c

```diff
@@ -942,13 +945,33 @@ int uv_pipe_listen(uv_pipe_t* handle, int backlog, uv_connection_cb cb) {
 }
 
 
+static DWORD WINAPI uv__pipe_zero_read_interrupter(void* param) {
+  HANDLE thread;
+  int r;
+  thread = (HANDLE) param;
+  for (;;) {
+    r = WaitForSingleObject(thread, PIPE_ZERO_READ_INTERRUPT_INTERVAL);
+    if (r == WAIT_TIMEOUT) {
+      /* ReadFile thread is active - interrupt ReadFile to give other */
+      /* processes a chance to act upon pipe */
+      pCancelSynchronousIo(thread);
+    } else {
+      /* ReadFile thread has terminated - cleanup the handle and exit */
+      CloseHandle(thread);
+      return 0;
+    }
+  }
+}
+
+
 static DWORD WINAPI uv_pipe_zero_readfile_thread_proc(void* parameter) {
   int result;
   DWORD bytes;
   uv_read_t* req = (uv_read_t*) parameter;
   uv_pipe_t* handle = (uv_pipe_t*) req->data;
   uv_loop_t* loop = handle->loop;
   HANDLE hThread = NULL;
+  HANDLE hThreadForInterrupt = NULL;
```

Comment:
I think the assignment should be separate, but clearly the surrounding code isn't doing that.